api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -27,16 +38,100 @@ public NameSupport() {
 	}
 	
 	public static NameSupport getInstance() {
-		if (singleton == null) {
-			throw new APIException("Not Yet Instantiated");
-		} else {
-			return singleton;
+		synchronized (NameSupport.class) {
```

Comment:
I'm not sure why we'd wrap this in a synchronized block instead of just marking the method as synchronized, if that's required. It has the same effect, but is easier to read.

Reply:
I think this was derived from [this other example](https://github.com/openmrs/openmrs-core/blob/d3ecfb38e9f82a680f821e0241f770a2cd720c38/api/src/main/java/org/openmrs/layout/address/AddressSupport.java#L43); I'd thought that would suffice...

Reply:
However, I don't think there's a real need for synchronization here at all. In AddressSupport, `getInstance()` creates the singleton instance. In `NameSupport`, we're relying on Spring to call the constructor. The thing that AddressSupport is synchronizing around is constructing the singleton instance.

Reply:
Ahh, cool then. I've gotten rid of `synchronized` block altogether.